### PR TITLE
Components: Filter picker: Add animation

### PR DESCRIPTION
**Pull request by psealock**
Create the animation in this video:
https://videopress.com/v/vNtQtCCs

This PR introduces the [react-transition-group](http://reactcommunity.org/react-transition-group/) to handle transitions between two states. It is needed because during the transition period previous and next states need to be rendered at the same time. This allows one to slide out while the other slides in.

## Test
1. Go to `http://localhost/wp-admin/admin.php?page=wc-admin#/analytics/products`
2. Click "All Products"
3. Click "Single Product"
4. Click "Single Product" to go back
5. Make sure the animation is the same as the video
6. Make sure the animation is smooth
7. Ensure the first element is focused after the animation has finished

### CODE REVIEW

File: client/components/animation-slider/index.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+			);
+			if ( focusable ) {
+				focusable.focus();
+			}
+		}
+	}
+
+	render() {
+		const { children, animationKey, animate } = this.props;
+		const args = { navRight: this.navRight, navLeft: this.navLeft };
```
Comment (timmyc) [inline client/components/animation-slider/index.js]: What do navRight/navLeft do here? seems to they both would be null/undefined and i don't see refs to it elsewhere in the code.

**Review by timmyc — COMMENTED** [review]
This is testing out wonderfully, very smooth nice animation! One minor question about the code but otherwise looking good to me.

File: client/components/animation-slider/index.js (hunk 1 above)
Comment (psealock) [reply client/components/animation-slider/index.js]: Oops, thanks. Thats leftover code from the previous attempt

**Review by timmyc — APPROVED** [review]
All ready to roll!

**Pull request merged**